scm/driver/gitea/webhook.go

```diff
@@ -190,27 +190,41 @@ func convertBranchHook(dst *createHook, action scm.Action) *scm.BranchHook {
 }
 
 func convertPushHook(dst *pushHook) *scm.PushHook {
-	return &scm.PushHook{
-		Ref: dst.Ref,
-		Commit: scm.Commit{
-			Sha:     dst.After,
-			Message: dst.Commits[0].Message,
-			Link:    dst.Compare,
-			Author: scm.Signature{
-				Login: dst.Commits[0].Author.Username,
-				Email: dst.Commits[0].Author.Email,
-				Name:  dst.Commits[0].Author.Name,
-				Date:  dst.Commits[0].Timestamp,
+	if len(hook.Commits) > 0 {
+		return &scm.PushHook{
+			Ref: dst.Ref,
+			Commit: scm.Commit{
+				Sha:     dst.After,
+				Message: dst.Commits[0].Message,
+				Link:    dst.Compare,
+				Author: scm.Signature{
+					Login: dst.Commits[0].Author.Username,
+					Email: dst.Commits[0].Author.Email,
+					Name:  dst.Commits[0].Author.Name,
+					Date:  dst.Commits[0].Timestamp,
+				},
+				Committer: scm.Signature{
+					Login: dst.Commits[0].Committer.Username,
+					Email: dst.Commits[0].Committer.Email,
+					Name:  dst.Commits[0].Committer.Name,
+					Date:  dst.Commits[0].Timestamp,
+				},
 			},
-			Committer: scm.Signature{
-				Login: dst.Commits[0].Committer.Username,
-				Email: dst.Commits[0].Committer.Email,
-				Name:  dst.Commits[0].Committer.Name,
-				Date:  dst.Commits[0].Timestamp,
+			Repo:   *convertRepository(&dst.Repository),
+			Sender: *convertUser(&dst.Sender),
+		}
+	} else {
+		return &scm.PushHook{
+			Ref: dst.Ref,
+			Commit: scm.Commit{
+				Sha:       dst.After,
+				Link:      dst.Compare,
+				Author:    scm.Signature{},
```

Comment:
maybe we use the `pusher` value here? https://github.com/drone/go-scm/blob/master/scm/driver/gitea/testdata/webhooks/push.json#L64:L70

Reply:
For author and commiter? Or only for one of them?

Reply:
I'm thinking for both ...